include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -133,6 +133,41 @@ namespace concurrency {
 class ThreadPoolParallelSection;
 class ThreadPoolLoop;
 
+/* Usage:
+1. In executor, call Start() before profiling and Stop() to get profiled numbers;
+2. Inside thread pool, call LogStart() before interested section and LogEnd... after to log elapsed time;
+3. To extend, just add more events in enum Event before "All", and update GetEventName(...) accordingly.
+4. Note LogStart must pair with either LogEnd or LogEndAndStart, otherwise ORT_ENFORCE will fail.
+*/
+class ThreadPoolProfiler {
+ public:
+  enum ThreadPoolEvent {
+    DISTRIBUTION = 0,
+    DISTRIBUTION_ENQUEUE,
+    RUN,
+    WAIT,
+    WAIT_REVOKE,
+    MAX_EVENT
+  };
+  ThreadPoolProfiler();
+  ~ThreadPoolProfiler() = default;
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(ThreadPoolProfiler);
+  using Clock = std::chrono::high_resolution_clock;
```

Comment:
>Clock [](start = 8, length = 5)

actually since onnxruntime::TimePoint is defined in common.h, how about defining onnxruntime::Clock there too? we can update TimePoint: `using TimePoint = Clock::time_point;`. seems easier to keep the two related types consistent that way.

Reply:
No, you can't use high_resolution_clock here. On Windows, probably it is fine, but on Linux, usually the class is implemented as:

```
clock_gettime(CLOCK_REALTIME, &tp);
```
This is the system-wide clock that measures real (i.e., wall-clock) time. You should use a monotonic clock instead. For example, CLOCK_MONOTONIC on Linux and QueryPerformanceCounter on Windows.

Reply:
see also std::chrono::steady_clock

---
In reply to: [585049908](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r585049908) [](ancestors = 585049908)

Reply:
looks like high_resolution_clock is still being used

---
In reply to: [585112307](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r585112307) [](ancestors = 585112307,585049908)